includes/reader-revenue/woocommerce/class-woocommerce-cli.php

```diff
@@ -162,6 +162,141 @@ function( $subscription ) {
 		}
 	}
 
+	/**
+	 * Fixes or reports active subscriptions that have missed next payment dates.
+	 * By default, will only process subscriptions started in the past 90 days.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--dry-run]
+	 * : If set, will report results but will not make any changes.
+	 *
+	 * [--batch-size=<batch-size>]
+	 * : The number of subscriptions to process in each batch. Default: 50.
+	 *
+	 * [--start-date=<date-string>]
+	 * : A date string in YYYY-MM-DD format to use as the start date for the script. Default: 90 days ago.
+	 *
+	 * @param array $args Positional args.
+	 * @param array $assoc_args Associative args.
+	 */
+	public function fix_missing_next_payment_dates( $args, $assoc_args ) {
+		$dry_run    = ! empty( $assoc_args['dry-run'] );
+		$now        = time();
+		$batch_size = ! empty( $assoc_args['batch-size'] ) ? intval( $assoc_args['batch-size'] ) : 50;
+		$start_date = ! empty( $assoc_args['start-date'] ) ? strtotime( $assoc_args['start-date'] ) : strtotime( '-90 days', $now );
+
+		\WP_CLI::log(
+			'
+Fetching active subscriptions with missing or missed next_payment dates...
+		'
+		);
+
+		$query_args    = [
+			'subscriptions_per_page' => $batch_size,
+			'subscription_status'    => [ 'active', 'pending' ],
+			'offset'                 => 0,
+		];
+		$processed     = 0;
+		$subscriptions = \wcs_get_subscriptions( $query_args );
+		$total_revenue = 0;
+		$results       = [];
+
+		while ( ! empty( $subscriptions ) ) {
+			foreach ( $subscriptions as $subscription_id => $subscription ) {
+				array_shift( $subscriptions );
+				$subscription_start = $subscription->get_date( 'start_date' );
+
+				// If the subscription start date is before the $args start date, we're done.
+				if ( strtotime( $subscription_start ) < $start_date ) {
+					$subscriptions = [];
+					break;
+				}
+
+				$next_payment_date = $subscription->get_date( 'next_payment' );
+				$is_in_past        = ! strtotime( $next_payment_date ) || strtotime( $next_payment_date ) < $now;
+
+				// Subscription has a valid next payment date and it's in the future, so skip.
+				if ( $next_payment_date && ! $is_in_past ) {
+					continue;
+				}
+
+				$result = [
+					'ID'                => $subscription->get_id(),
+					'status'            => $subscription->get_status(),
+					'start_date'        => $subscription_start,
+					'next_payment_date' => $next_payment_date,
+					'billing_period'    => $subscription->get_billing_period(),
```

Comment:
Because `'billing_interval'` is never set in the `$result`, all items will bail on the `if` on line 235. 

```suggestion
					'billing_period'    => $subscription->get_billing_period(),
					'billing_interval'  => $subscription->get_billing_interval(),
```

Reply:
Whoops, you're right! Sorry for this oversight.